### PR TITLE
chore(docs): add migration documentation

### PR DESCRIPTION
**Pull request by hzhu**
## Description

This pull request adds the documentation for migration from v7 to v8.

### CODE REVIEW

File: docs/migration.md [hunk 1]
```diff
@@ -0,0 +1,299 @@
+# Garden v8 Migration Guide
```
Comment (austin94) [inline docs/migration.md]: This is going to be a continuous migration guide, not just for v8. We may want to make the title `Garden migration guide` and have an `h2` as `v8`. All the other headings should be incremented as well.

File: docs/migration.md [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+
+The upgrade to Garden v8 is designed to allow you to upgrade each package individually. It is not
+necessary to upgrade all Garden packages to v8 at once. Each upgrade step can be completed in
+isolation.
+
+### Upgrade Steps
+
+- Upgrade all existing Garden React dependencies to their [latest `v7`](https://github.com/zendeskgarden/react-components/blob/master/CHANGELOG.md#v700-2019-10-17)
+  version.
+
```
Comment (austin94) [inline docs/migration.md]: Why is there an extra space here?

File: docs/migration.md [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+### @zendeskgarden/react-forms
+
+- The following props have been renamed or removed:
+  - The following props have been removed from all components:
+    - `focused`
+    - `hovered`
+    - `mediaLayout`
+    - `open`
+    - `select
```
Comment (austin94) [inline docs/migration.md]: Missing `

File: docs/migration.md [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+    - `tagLayout`
+  - Field
+    - `inline` -> removed
+      - Inline layouts can now be achieved with custom styling.
+  - Hint
+    - Small` -> removed
```
Comment (austin94) [inline docs/migration.md]: Missing `

File: docs/migration.md [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+      - Use `isCompact` on the `Input` component instead.
+  - Label
+    - `regular ->`isRegular`
```
Comment (austin94) [inline docs/migration.md]: Missing `

File: docs/migration.md [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+
+### @zendeskgarden/react-pagination
+
+- The following components have been removed:
+  - `Gap`
+  - `NextPage`
+  - `Page`
+  - `PaginationView`
+  - `PreviousPage`
+- A new `pageGap` prop has been added - It is used to modify the gap display positioning
```
Comment (austin94) [inline docs/migration.md]: Should this description be another bulleted item?

File: docs/migration.md [hunk 7]
```diff
@@ -0,0 +1,299 @@
...
+```jsx
+<Tabs>
+  <TabList>
+    <Tab item="tab-1">Tab 1</Tab>
+  </TabList>
+  <TabPanel item="tab-1">Content</TabPanel>
+</Tabs>
+```
+
+- All state and styling props such as `active`, `focused`, ‘hover`,`selected`, etc. have been removed.
```
Comment (austin94) [inline docs/migration.md]: Missing ` characters

File: docs/migration.md [hunk 8]
```diff
@@ -0,0 +1,304 @@
...
+### Breaking Changes
+
+#### All packages
+
+- Garden v8 packages use CSS-in-JS and no longer provide CSS files.
+  - Remove all Garden React CSS imports.
+  - i.e `@zendeskgarden/react-buttons/dist/styles.css`
+- All themable component IDs (`data-garden-id` attributes) have been renamed for consistency across
+  all packages.
+- Any custom theming provided to the `ThemProvider` component should be checked for accuracy.
```
Comment (hzhu) [inline docs/migration.md]: Typo.

**Pull request merged**